### ISSUE TITLE
make run crashed on Ubuntu

### ISSUE THREAD

**Issue by gkorland** [lantiga/RedisTF#10]
I'm getting the following error when trying to run 

```
make run
```

```
=== REDIS BUG REPORT START: Cut & paste starting from here ===
20321:M 13 Sep 2018 12:18:05.167 # Redis 999.999.999 crashed by signal: 11
20321:M 13 Sep 2018 12:18:05.167 # Crashed running the instruction at: 0x7f11649ae244
20321:M 13 Sep 2018 12:18:05.167 # Accessing address: 0x8058
20321:M 13 Sep 2018 12:18:05.167 # Failed assertion: <no assertion failed> (<no file>:0)

------ STACK TRACE ------
EIP:
/lib/x86_64-linux-gnu/libpthread.so.0(__pthread_mutex_trylock+0x14)[0x7f11649ae244]

Backtrace:
../install//redis-server *:6379(logStackTrace+0x5a)[0x562915eb1caa]
../install//redis-server *:6379(sigsegvHandler+0xb1)[0x562915eb2461]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12890)[0x7f11649b6890]
/lib/x86_64-linux-gnu/libpthread.so.0(__pthread_mutex_trylock+0x14)[0x7f11649ae244]
../install//redis-server *:6379(je_base_alloc+0x51)[0x562915f2dfe1]
../install/libtensorflow_framework.so(+0x3db2b8)[0x7f115db3a2b8]
../install/libtensorflow_framework.so(+0x3e1cbc)[0x7f115db40cbc]
/lib64/ld-linux-x86-64.so.2(+0x10733)[0x7f116537d733]
/lib64/ld-linux-x86-64.so.2(+0x151ff)[0x7f11653821ff]
/lib/x86_64-linux-gnu/libc.so.6(_dl_catch_exception+0x6f)[0x7f116471a2df]
/lib64/ld-linux-x86-64.so.2(+0x147ca)[0x7f11653817ca]
/lib/x86_64-linux-gnu/libdl.so.2(+0xf96)[0x7f1164dcbf96]
/lib/x86_64-linux-gnu/libc.so.6(_dl_catch_exception+0x6f)[0x7f116471a2df]
/lib/x86_64-linux-gnu/libc.so.6(_dl_catch_error+0x2f)[0x7f116471a36f]
/lib/x86_64-linux-gnu/libdl.so.2(+0x1735)[0x7f1164dcc735]
/lib/x86_64-linux-gnu/libdl.so.2(dlopen+0x71)[0x7f1164dcc051]
../install//redis-server *:6379(moduleLoad+0x4a)[0x562915ee0a3a]
../install//redis-server *:6379(moduleLoadFromQueue+0x43)[0x562915ee0bb3]
../install//redis-server *:6379(main+0x470)[0x562915e62bc0]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7)[0x7f11645d4b97]
../install//redis-server *:6379(_start+0x2a)[0x562915e62e9a]
```

**Comment by lantiga** [comment]
Hi @gkorland, I am indeed reproducing this on a ubuntu box. Investigating right now.

**Comment by lantiga** [comment]
I have a workaround.

I started by noticing that an empty redis module crashes upon loading the moment you add `-ltensorflow` to the `LD_FLAGS`.

Inspecting the stack trace I noticed that the crash was related to `jemalloc`, so I built Redis with `libc` malloc as the allocator. This prevents the crash and `make run` succeeds.

I'll need to understand more, but for the time being I'll set the `MALLOC` env var for building Redis in `get_deps.sh`.
